pkg/codegen/internal/test/testdata/aws-s3-folder.pp.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"mime"
+	"path"
+
+	"github.com/pulumi/pulumi-aws/sdk/v2/go/aws/s3"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		siteBucket, err := s3.NewBucket(ctx, "siteBucket", &s3.BucketArgs{
+			Website: &s3.BucketWebsiteArgs{
+				IndexDocument: pulumi.String("index.html"),
+			},
+		})
+		if err != nil {
+			return err
+		}
+		siteDir := "www"
+		files0, err := ioutil.ReadDir(siteDir)
+		if err != nil {
+			return err
+		}
+		fileNames0 := make([]string, len(files0))
+		for i0, val0 := range files0 {
+			fileNames0[i0] = val0.Name()
+		}
+		for i0, val0 := range fileNames0 {
+			_, err = s3.NewBucketObject(ctx, "files-"+string(i0), &s3.BucketObjectArgs{
+				Bucket:      siteBucket.ID(),
+				Key:         pulumi.String(val0),
+				Source:      pulumi.NewFileAsset(fmt.Sprintf("%v%v%v", siteDir, "/", val0)),
```

Comment:
I realize that this is a result of the input program, but paths should really use `path.Join` instead of a hardcoded slash.

Reply:
Agreed, but there isn't really a good way to handle this (at least in terms of a sensible effort/reward ratio). The other language generators do the same.